bundles/amazon/ack/elasticache/bundle/config/01-replication-group.ytt.yml

```diff
@@ -0,0 +1,225 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:assert", "assert")
+
+#@ def get_id(x): return "{}-{}".format(data.values.name, x)
+
+#@ def tags(resource_tags=[]):
+#@   return [ x for x in (data.values.tags + resource_tags) if x["key"] != "" and x["value"] != "" ]
+#@ end
+
+#@ users = {
+#@   "reader": { "access_string": "on ~* -@all +@read" },
+#@   "writer": { "access_string": "on ~* +@all" },
+#@ }
+
+#! only Redis is currently supported
+#@ engine = "redis"
+
+#@ replicationGroupName = data.values.name
+#@ cacheParameterGroupName = data.values.name
+
+#@ service_account_name = "{}-{}".format(data.values.name, "elasticache-reader")
+#@ role_name = service_account_name
+
+#! a user with name "default" must be present in every usergroup
+#! the elasticache "default" user has all permissions and no password -> unsuitable
+#! this creates a new "default" user with no permissions
+#@ default_user = { "id": get_id("default"), "access_string": "off ~* -@all" }
+
+#@ mandatory_fields = [ "name", "namespace", "cacheSubnetGroupName", "cacheNodeType", "vpcSecurityGroupIDs" ]
+#@ missing_fields = []
+#@ for field in mandatory_fields:
+#@   if len(data.values[field]) == 0:
+#@     missing_fields.append(field)
+#@   end
+#@ end
+
+#@ if len(missing_fields) > 0:
+#@   assert.fail("Missing values for mandatory fields '{}'".format(missing_fields))
+#@ end
+
+#@ if data.values.createNamespace and data.values.namespace:
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: #@ data.values.namespace
+#@ end
+---
+apiVersion: elasticache.services.k8s.aws/v1alpha1
+kind: User
+metadata:
+  name: #@ default_user["id"]
+  namespace: #@ data.values.namespace
+spec:
+  accessString: #@ default_user["access_string"]
+  noPasswordRequired: true
+  engine: #@ engine
+  userID: #@ default_user["id"]
+  userName: default
+  tags: #@ tags()
+
+#@ for user_name in users:
+#@   user_id = get_id(user_name)
+#@   user_creds = "{}-{}".format(user_id,"creds")
+#@   users[user_name].update({ "id": user_id, "creds": user_creds })
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: Password
+metadata:
+  name: #@ user_creds
+  namespace: #@ data.values.namespace
+spec:
+  length: 128
+  secretTemplate:
+    type: Opaque
+    stringData:
+      password: $(value)
+---
+apiVersion: secretgen.carvel.dev/v1alpha1
+kind: SecretTemplate
+metadata:
+  name: #@ "{}-bindable".format(user_creds)
+  namespace: #@ data.values.namespace
+spec:
+  serviceAccountName: #@ service_account_name
+  inputResources:
+  - name: replicationGroup
+    ref:
+      apiVersion: elasticache.services.k8s.aws/v1alpha1
+      kind: ReplicationGroup
+      name: #@ data.values.name
+  - name: creds
+    ref:
+      apiVersion: v1
+      kind: Secret
+      name: #@ user_creds
+  template:
+    metadata:
+      labels:
+        services.apps.tanzu.vmware.com/class: redis-small
+    type: servicebinding.io/redis
```

Comment:
**suggestion** Not part of this PR, but it would be great to fix the other bundles we have. Right nwo they do not have the `servicebinding.io` prefix. E.g. https://github.com/vmware-tanzu/tanzu-application-platform-reference-service-packages/blob/main/bundles/amazon/ack/rds/bundle/config/config.yaml#L75

Reply:
Correct. We also have to fix the folder names, like making `amazon` become `aws`. 
We'll have a separate piece of work and PR for this.